### PR TITLE
add searchbar component, functionality and styling

### PR DESCRIPTION
**Pull request by eduskx**
Thank you very much for having a look 🫶🏻

### CODE REVIEW

File: components/SearchBar/index.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import styled from "styled-components";
+
+export default function SearchBar({ setSearchInput }) {
+  function inputHandler(event) {
+    const lowerCasedInput = event.target.value.toLowerCase();
+    setSearchInput(lowerCasedInput);
+  }
```
Comment (NikRoe) [inline components/SearchBar/index.js:7]: I would move this function one level higher.

In general it is a good practice to keep the setter functions of a state close to where you initialize the state.
It keeps your code cleaner and also easier to maintain.

My suggestion has a few adaptations as its consequence:

```suggestion

```

File: components/SearchBar/index.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import styled from "styled-components";
+
+export default function SearchBar({ setSearchInput }) {
```
Comment (NikRoe) [inline components/SearchBar/index.js:3]:
```suggestion
export default function SearchBar({ onSearch }) {

```

File: components/SearchBar/index.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+  return (
+    <StyledContainer>
+      <label htmlFor="searchbar"></label>
+      <StyledSearchBar
+        type="text"
+        id="searchbar"
+        name="searchbar"
+        placeholder="🔎 Search projects"
+        onChange={inputHandler}
```
Comment (NikRoe) [inline components/SearchBar/index.js:17]:
```suggestion
        onChange={onSearch}

```

File: pages/index.js [hunk 4]
```diff
@@ -22,7 +26,8 @@ export default function HomePage({ projects, setNewProjects }) {
         formSteps={formSteps}
         setFormSteps={setFormSteps}
       />
-      <FilterList projects={projects} />
+      <SearchBar setSearchInput={setSearchInput} />
```
Comment (NikRoe) [inline pages/index.js:29]:
```suggestion
      <SearchBar onSearch={handleSearch} />
```

File: pages/index.js [hunk 5]
```diff
@@ -2,8 +2,12 @@ import Form from "@/components/Form";
 import Header from "@/components/Header";
 import useLocalStorageState from "use-local-storage-state";
 import FilterList from "@/components/FilterList";
+import SearchBar from "@/components/SearchBar";
+import { useState } from "react";
 
 export default function HomePage({ projects, setNewProjects }) {
+  const [searchInput, setSearchInput] = useState("");
+
```
Comment (NikRoe) [inline pages/index.js:10]:
```suggestion
function handleSearch(event) {
    const lowerCasedInput = event.target.value.toLowerCase();
    setSearchInput(lowerCasedInput);
  }
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi team,

nicely done 🙌 

However, I recommend working with a helper function to set the searchInput state and write this function in the same file where your state lives.

This is in general a good approach for several reasons:

1. Better Control Over Logic
By writing helper functions, you can centralize and control the logic executed when the state changes. This allows you to add additional validations, transformations, or other logical operations before the state is actually changed.

2. Avoiding Errors
Directly passing the setter function can lead to unintended and hard-to-debug errors when the state is modified from multiple places. Helper functions ensure that the state is changed in a predictable and controlled manner.

3. Increased Readability and Maintainability
Helper functions increase the readability and maintainability of the code because they perform clearly named and specific tasks. This makes it easier for other developers (and for yourself in the future) to understand and extend the code.

4. Encapsulation of State Logic
Helper functions enable you to separate state logic from UI logic. This leads to better encapsulation and more modular code, which is easier to test and maintain.

5. Reusability
Helper functions can be reused if they are well-abstracted and generalized. This reduces redundancy and promotes consistency within the application.

Long story short: By implementing my suggestions, your app should still work just as fine as before, while having all of the above benefits 🙏

**Comment by eduskx** [comment]
Thank you for this detailed explanation! Our team loves your reviews :)

**Review by NikRoe — APPROVED** [review]
LGTM
